pkg/resources/grant_helpers.go

```diff
@@ -147,22 +147,58 @@ func readGenericGrant(data *schema.ResourceData, meta interface{}, builder snowf
 	}
 	priv := data.Get("privilege").(string)
 
-	var roles, shares []string
+	// Map of roles to privileges
+	rolePrivileges := map[string]map[string]struct{}{}
```

Comment:
I think if you are going to use a map as a set (which is reasonable) having `map[string]bool` might be clearer.

Reply:
I think it is better just because it is easier to understand/read. Also using a type alias probably makes sense here.

Example https://play.golang.org/p/kdViq4Mg0Pn